olp-cpp-sdk-dataservice-read/tests/TaskContextTest.cpp

```diff
@@ -0,0 +1,242 @@
+/*
+ * Copyright (C) 2019 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+#include "TaskContext.h"
+
+#include <gtest/gtest.h>
+
+namespace {
+
+using namespace olp::client;
+using namespace olp::dataservice::read;
+
+using ResponseType = std::string;
+using Response = ApiResponse<ResponseType, ApiError>;
+using ExecuteFunc = std::function<Response(CancellationContext)>;
+using Callback = std::function<void(Response)>;
+
+class TestableTaskContext : public TaskContext {
```

Comment:
```suggestion
class TaskContextTestable : public TaskContext {
```
As I remember that we agreed to use `Testable` as suffix.